core/src/main/java/com/valtech/aem/saas/core/fulltextsearch/ConfiguredFiltersParser.java

```diff
@@ -0,0 +1,45 @@
+package com.valtech.aem.saas.core.fulltextsearch;
+
+import com.valtech.aem.saas.api.fulltextsearch.FilterModel;
+import com.valtech.aem.saas.api.query.Filter;
+import com.valtech.aem.saas.api.query.FilterFactory;
+import java.util.Arrays;
+import java.util.List;
+import java.util.Optional;
+import java.util.Set;
+import java.util.stream.Collectors;
+import java.util.stream.Stream;
+import lombok.RequiredArgsConstructor;
+import org.apache.commons.lang3.StringUtils;
+
+@RequiredArgsConstructor
+public class ConfiguredFiltersParser {
```

Comment:
why don't we use a traditional sling model for the parsing?

Reply:
I moved the parsing/mapping logic in the FilterModelmpl sling model.
Maybe one reason why a helper class seems ok to me, is the light weight unit test (avoid usage of aem mocks)